icu4j/main/core/src/main/java/com/ibm/icu/segmenter/SegmentsImpl.java

```diff
@@ -40,6 +40,10 @@ public Segment segmentAt(int i) {
     int start;
     int limit;
 
+    if (i < 0 || i >= breakIterPrototype.last()) {
```

Comment:
Calling breakIterPrototype.last() is not thread-safe.
Please just use source.length() here.

Reply:
Done.